pennylane/tape/tapes/reversible.py

```diff
@@ -158,9 +171,10 @@ def analytic_pd(self, idx, device, params=None, **options):
                 "reversible gradient method.".format(op.name)
             )
 
-        if self._state is None:
-            self.execute_device(params, device)
-            self._state = device._pre_rotated_state
+        # get the stored final state of the original circuit, from which we start here
+        final_state = self._final_state
+        # get the number of wires on the device used for the differentiation

```

Comment:
are we getting the number of wires, or the wires themselves?

Reply:
@mariaschuld?